tests/Traits/DataHelpers/SalesChannelHelper.php

```diff
@@ -61,19 +59,26 @@ public function buildSalesChannelContext(
         return $salesChannelContextFactory->create(
             Uuid::randomHex(),
             $salesChannelId,
-            $this->buildSalesChannelContextFactoryOptions($customerEntity, $languageId)
+            $this->buildSalesChannelContextFactoryOptions($customerEntity, $languageId, $currencyId)
         );
     }
 
-    private function buildSalesChannelContextFactoryOptions(?CustomerEntity $customerEntity, ?string $languageId): array
-    {
+    private function buildSalesChannelContextFactoryOptions(
+        ?CustomerEntity $customerEntity,
+        ?string $languageId,
+        ?string $currencyId
+    ): array {
         $options = [];
         if ($customerEntity) {
             $options[SalesChannelContextService::CUSTOMER_ID] = $customerEntity->getId();
         }
         if ($languageId) {
             $options[SalesChannelContextService::LANGUAGE_ID] = $languageId;
         }
+        // Currency is not available in the sales channel context below Shopware v6.4.
+        if ($currencyId && !Utils::versionLowerThan('6.4')) {
```

Comment:
The currency id already exists since the first Shopware 6 version. https://github.com/shopware/platform/blob/v6.0.0+dp1/src/Core/System/SalesChannel/Context/SalesChannelContextService.php. The difference is that the helper is only available since a later version, so you need to get the SalesChannel and assign the currency id to the sales channel.